lib/live-stream.js

```diff
@@ -31,34 +29,57 @@ function streamData (args, archives, cb) {
       delete archives[link]
       cb()
       dat.listStream.destroy()
-      if (dat.path.indexOf(root) > -1) {
-        rmrf(dat.path, err => {
-          if (err) throw err
+      dat.close(function (err) {
+        if (err) console.error(err)
+        dat.db.close(function (err) {
+          if (err) console.error(err)
+          if (dat.path.indexOf(root) > -1) {
+            rmrf(dat.path, err => {
+              if (err) console.error(err)
+            })
+          }
         })
-      }
+      })
     } else {
+      if (archives[link]) return
       const path = `${root}/${link}`
       fs.mkdir(path, () => {
-        const archive = args.createArchive(args.drive, Object.assign({ key }, data.value))
-        archive.open(cb)
-        archive.swarm = swarm(archive)
-        archive.swarm.on('connection', peer => {
-          cb()
-          peer.on('close', () => cb())
-        })
-        archive.on('download', () => cb())
-        archive.on('content', () => cb())
-        archive.listStream = archive.list({ live: true })
-        archive.listStream.on('data', entry => {
-          if (entry.name !== 'dat.json') return
-          collect(archive.createFileReadStream('dat.json'), (err, raw) => {
-            if (err) return
-            const json = JSON.parse(raw.toString())
-            archive.title = json.title
-            cb()
+        const archive = args.createArchive(Object.assign({ key }, data.value))
```

Comment:
agree that this is a bit hard to read; perhaps using a bit less ES6 could help with this?